pkg/asset/machines/master_test.go

```diff
@@ -100,6 +100,55 @@ spec:
   osImageURL: ""
 `
 
+var AROEtcHostsMasterMachineConfig = `apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  creationTimestamp: null
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 99-master-aro-etc-hosts-gateway-domains
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,CWFwaS50ZXN0LWNsdXN0ZXIudGVzdC1kb21haW4gYXBpLWludC50ZXN0LWNsdXN0ZXIudGVzdC1kb21haW4KCQo=
```

Comment:
It'll be a bit more work, but can we build up this source rather than putting the base64 literal here, so the test can ensure exactly the gateway domains we expect to see end up in the file?